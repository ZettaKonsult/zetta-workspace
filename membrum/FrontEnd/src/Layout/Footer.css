.Footer {
  background-color: dimgrey;
  color: white;
  display: flex;
  flex-flow: row nowrap;
  flex: 0 0 auto;
}

.FooterCopyRight {
  user-select: none;
  margin-left: 0.5em;
  align-self: center;
}
.FooterContent {
  flex: 1 0 auto;
}

.FooterSocialMedia {
  align-self: flex-end;
  color: black;
}

.FooterSocialMediaIcon {
  align-self: flex-end;
  color: black;
}
