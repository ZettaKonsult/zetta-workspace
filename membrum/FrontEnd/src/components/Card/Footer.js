import React from 'react'

const Footer = props => (
  <div className="CardFooter">
    <span>View Details</span>
    <span>
      <i className="fa fa-arrow-circle-right" />
    </span>
  </div>
)

Footer.defaultProps = {
  link: '/'
}
export default Footer
