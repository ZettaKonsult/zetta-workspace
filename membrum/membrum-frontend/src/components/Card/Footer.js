import React from "react"
import { Link } from "react-router-dom"

const Footer = props => (
  <Link to={props.link}>
    <div className="CardFooter">
      <span>View Details</span>
      <span>
        <i className="fa fa-arrow-circle-right" />
      </span>
    </div>
  </Link>
)

Footer.defaultProps = {
  link: "/"
}
export default Footer
