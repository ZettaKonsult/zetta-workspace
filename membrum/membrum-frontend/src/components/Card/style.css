:root {
  --theme-color: var(--primary);
}

.Card {
  display: flex;
  flex-flow: column nowrap;
}

.Card a {
  text-decoration: none;
  color: black;
}

.Card a:hover {
  color: var(--primary);
}

.CardContent {
  background-color: #337ab7;
  background-color: var(--theme-color);
  border: 1px solid var(--theme-color);
  border-radius: 5px 5px 0 0;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0.5em;
  color: white;
}

.CardFooter {
  background-color: #f5f5f5;
  border: 1px solid var(--theme-color);
  border-radius: 0 0 5px 5px;
  display: flex;
  justify-content: space-between;
  padding: 0.5em;
  font-size: 1.2em;
}

.CardText {
  text-align: right;
  font-size: 1em;
}
.CardText .CardValue {
  font-size: 2.5em;
}
