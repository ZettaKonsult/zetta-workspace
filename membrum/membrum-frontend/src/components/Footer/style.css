.Footer {
  background-color: black;
  color: white;
  display: flex;
  flex-flow: row nowrap;
  flex: 0 0 auto;
}

.FooterCopyRight {
  align-self: center;
}
.FooterContent {
  flex: 1 0 auto;
}

.FooterSocialMedia {
  align-self: flex-end;
  color: black;
}

.FooterSocialMediaIcon {
  align-self: flex-end;
  color: black;
}
