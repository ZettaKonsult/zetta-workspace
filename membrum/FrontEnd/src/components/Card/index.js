import React from 'react';
import PropTypes from 'prop-types';

import Footer from './Footer';

import './style.css';

const Card = props => (
  <div className="Card">
    <div
      style={{ '--theme-color': `var(--${props.type})` }}
      className="CardContent"
    >
      <i className={'CardIcon fa fa-5x fa-' + props.icon} />
      <div className="CardText">
        <div className="CardValue">{props.value}</div>
        <div>{props.label}</div>
      </div>
    </div>
    <Footer type={props.type} link={props.link} />
  </div>
);

Card.defaultProps = {
  type: 'primary',
  label: 'New Comments',
  icon: 'comments',
  value: '24',
};

Card.propTypes = {
  type: PropTypes.oneOf(['primary', 'warning', 'danger', 'info', 'success']),
  label: PropTypes.string,
  icon: PropTypes.oneOf(['comments', 'tasks', 'shopping-cart', 'support']),
  value: PropTypes.string,
};

export default Card;
