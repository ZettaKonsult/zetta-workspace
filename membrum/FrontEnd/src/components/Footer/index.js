import React from "react"

import "./style.css"

const Footer = () => (
  <div className="Footer">
    <div className="FooterCopyRight">&copy;{new Date().getFullYear()}</div>
    <div className="FooterContent" />
    <div className="FooterSocialMedia">
      <a
        className="FooterSocialMediaIcon"
        href="https://www.facebook.com/Studentlund/"
      >
        <span className="fa-stack fa-2x ">
          <i className="fa fa-circle fa-stack-2x" />
          <i className="fa fa-facebook fa-inverse fa-stack-1x" />
        </span>
      </a>
      <a
        className="FooterSocialMediaIcon"
        href="https://twitter.com/studentlund?lang=en"
      >
        <span className="fa-stack fa-2x ">
          <i className="fa fa-circle fa-stack-2x" />
          <i className="fa fa-twitter fa-inverse fa-stack-1x" />
        </span>
      </a>
      <a
        className="FooterSocialMediaIcon"
        href="https://www.instagram.com/studentlund/"
      >
        <span className="fa-stack fa-2x ">
          <i className="fa fa-circle fa-stack-2x" />
          <i className="fa fa-instagram fa-inverse fa-stack-1x" />
        </span>
      </a>
    </div>
  </div>
)

export default Footer
