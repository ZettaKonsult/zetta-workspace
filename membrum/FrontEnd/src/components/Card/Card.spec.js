import React from 'react'
import renderer from 'react-test-renderer'

import Card from './'
import Footer from './Footer'

it('should render a Card', function() {
  const tree = renderer.create(<Card />).toJSON()
  expect(tree).toMatchSnapshot()
})
