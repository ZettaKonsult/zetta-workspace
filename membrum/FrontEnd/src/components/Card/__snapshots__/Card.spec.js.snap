// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render a Card 1`] = `
<div
  className="Card"
>
  <div
    className="CardContent"
    style={
      Object {
        "--theme-color": "var(--primary)",
      }
    }
  >
    <i
      className="CardIcon fa fa-5x fa-comments"
    />
    <div
      className="CardText"
    >
      <div
        className="CardValue"
      >
        24
      </div>
      <div>
        New Comments
      </div>
    </div>
  </div>
  <div
    className="CardFooter"
  >
    <span>
      View Details
    </span>
    <span>
      <i
        className="fa fa-arrow-circle-right"
      />
    </span>
  </div>
</div>
`;
